numpy/core/src/multiarray/common.c

```diff
@@ -945,5 +945,24 @@ new_array_for_sum(PyArrayObject *ap1, PyArrayObject *ap2, PyArrayObject* out,
     }
 }
 
-
-
+bool
+has_rank_0(PyArrayObject *v)
```

Comment:
Probably better to call this
```suggestion
check_has_rank_0(PyArrayObject *v)
```
and give it a `int` return of either `0` or `-1`, to make it clear that it sets an error